HelloVectors/HelloVectors_win32_cpp/LottieTool/SquareCircleMorph.cpp

```diff
@@ -0,0 +1,355 @@
+//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by a tool.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+#include "pch.h"
+#include "SquareCircleMorph.h"
+#include "d2d1.h"
+#include <d2d1_1.h>
+#include <d2d1helper.h>
+
+#include <Windows.Graphics.Interop.h>
+
+using namespace winrt;
+using namespace Windows::UI::Composition;
+using namespace Windows::Graphics;
+
+namespace
+{
+	class GeoSource final :
+		public ABI::Windows::Graphics::IGeometrySource2D,
+		public ABI::Windows::Graphics::IGeometrySource2DInterop
+	{
+		ULONG _cRef;
+		com_ptr<ID2D1Geometry> _cpGeometry;
+
+	public:
+		GeoSource(com_ptr<ID2D1Geometry> pGeometry)
+			: _cRef(1)
+			, _cpGeometry(pGeometry)
+		{ }
+
+		IFACEMETHODIMP QueryInterface(REFIID iid, void ** ppvObject) override
+		{
+			if (iid == __uuidof(ABI::Windows::Graphics::IGeometrySource2DInterop))
+			{
+				AddRef();
+				*ppvObject = static_cast<ABI::Windows::Graphics::IGeometrySource2DInterop*>(this);
+				return S_OK;
+			} else if (iid == __uuidof(ABI::Windows::Graphics::IGeometrySource2D))
+			{
+				AddRef();
+				*ppvObject = static_cast<ABI::Windows::Graphics::IGeometrySource2D*>(this);
+				return S_OK;
+			}
+			return E_NOINTERFACE;
+		}
+
+		IFACEMETHODIMP_(ULONG) AddRef() override
+		{
+			return InterlockedIncrement(&_cRef);
+		}
+
+		IFACEMETHODIMP_(ULONG) Release() override
+		{
+			ULONG cRef = InterlockedDecrement(&_cRef);
+			if (cRef == 0)
+			{
+				delete this;
+			}
+			return cRef;
+		}
+
+		IFACEMETHODIMP GetIids(ULONG*, IID**) override
+		{
+			return E_NOTIMPL;
```

Comment:
This is a crash waiting to happen. All out params must be set/cleared. Better to use winrt::implements and get this for free.

Reply:
Fixed in next PR